services/question/README.md

```diff
@@ -0,0 +1,103 @@
+# Question Service User Guide
+
+## Get All Questions
+This endpoint allows the retrieval of all the questions in the database.
+
+- **HTTP Method**: `GET`
+- **Endpoint**: `/questions`
+
+### Responses:
+
+| Response Code | Explanation                                         |
+|---------------|-----------------------------------------------------|
+| 200 (OK)      | Success, all questions are returned                 |
+| 500 (Internal Server Error) | Unexpected error in the database or server |
+
+**Example of Response Body for Success**:  
+TBC
+
+## Get Question by ID
+This endpoint allows the retrieval of the question by using the question ID.
+
+- **HTTP Method**: `GET`
+- **Endpoint**: `/questions/{questionId}`
+
+### Parameters:
+
+- Required: `questionId` path parameter
+
+### Responses:
+
+| Response Code | Explanation                                         |
+|---------------|-----------------------------------------------------|
+| 200 (OK)      | Success, question corresponding to the questionID is returned |
+| 404 (Not Found) | Question with the specified questionID not found   |
+| 500 (Internal Server Error) | Unexpected error in the database or server |
+
+**Example of Response Body for Success**:  
+TBC
+
+## Get Question by Parameters
+This endpoint allows the retrieval of a random question that matches the parameters provided.
+
+- **HTTP Method**: `GET`
+- **Endpoint**: `/questions/search`
+
+### Parameters:
+
+- `limit` - The number of questions to be returned (Required)
+- `topics` - The topic of the question (Required)
+- `languages` - The language of the question (Required)
```

Comment:
Does this mean that we are going to customize the code template for each question? Personally, I'm not too keen on doing so, and hence I don't think we need this field.